### PR TITLE
Add Qwen3 Eagle3 first layer support

### PR DESCRIPTION
**Pull request by guan404ming**
## Why

Eagle3 model_definitions only registered Llama, leaving Qwen3 unsupported despite base_components declaring it.

## How

- Add `Qwen3DecoderEagle3FirstLayer` with 2x hidden_size q/k/v projections and separate embed/hidden normalization
- Register `"qwen3"` entry in Eagle3 `model_classes` via `override_components`

### CODE REVIEW

File: src/speculators/models/eagle3/model_definitions.py [hunk 1]
```diff
@@ -90,8 +92,93 @@ def forward(
...
+        mid = hidden_states.shape[2] // 2
+        embeds, hidden = hidden_states.split(mid, dim=-1)
+        residual = hidden
+
+        # Apply norms
+        embeds = self.input_layernorm(embeds)
+        hidden = self.hidden_norm(hidden)
+        if self.norm_before_residual:
+            residual = hidden  # set residual to normalized hidden
+        hidden_states = torch.cat([embeds, hidden], dim=-1)
```
[the remark this answers is not in the document]
Comment (guan404ming) [reply src/speculators/models/eagle3/model_definitions.py]: I think this one makes sense. Just updated!

**Comment by guan404ming** [comment]
Thanks! Please feel free to let me know if there’s anything that needs to be updated before merging this.

By the way, I’m currently planning to contribute more regularly to the vLLM project. I was wondering if you have any recommendations for areas to contribute to, either in this project or others. Also, are there any upcoming plans or tasks where I might be able to help out?

Thanks again!

**Comment by guan404ming** [comment]
Thanks for the update and for merging the PR!

Those new features sound awesome. I'm currently doing research about speculative decoding and vLLM integration lately so I'm super interested in where this is heading. I'm happy to dive in on the smaller tasks or testing while those refactor is happening. Feel free to ping me anytime. I'll keep an eye on the new issues in the coming weeks. Cheers!
